server/models/user.ts

```diff
@@ -0,0 +1,86 @@
+"use strict";
+import Sequelize, { Model, UUIDV4, DataTypes } from "sequelize";
+
+
+interface UserAttributes {
+  /* Login Specific */
+  id: string;
+  userName: string;
+  password: string; // hash
+  email: string;
+
+  /* Logs */
+  lastLogin: Date;
+  karma: Number;
+};
+
+
+module.exports = (sequelize: Sequelize.Sequelize) => {
+
+  class User extends Model<UserAttributes> implements UserAttributes {
+    id!: string; // uuidv4
+    userName!: string;
+    password!: string;
+    email!: string;
+
+    lastLogin!: Date;
+    karma!: Number; // Should not be revealed to public
+
+    static associate(model: any){
+      User.hasMany(model.Post); 
+      User.hasMany(model.Comment);
+      /* Userid is stored inside of respective Post, and Comment */
+    }
+
+  };
+
+   User.init({
+    id:
+    {
+      type: DataTypes.UUID,
+      defaultValue: UUIDV4,
+      allowNull: false,
+      primaryKey: true,
+      unique: true
+    },
+    
+    userName:
+    {
+      type: DataTypes.STRING(32),
+      allowNull: false,
+      unique: true
+    },    
+    
+    password:
+    {
+      type: DataTypes.STRING,
+      allowNull: false
+    },   
+    email:
+    {
+      type: DataTypes.STRING(64),
+      allowNull: false,
+      unique: true,
+      validate : {
+         is: [".[a-zA-z\-\_\.0-9]*@[a-zA-z\-\_\.0-9]*utoronto.ca"],  // only utor emails
```

Comment:
Lets write some tests for this.

Reply:
Sure. I will write database tests and make a separate PR after

Reply:
Can you do it in this one? It's better to keep the tests and the code they test on a single pr.

Reply:
Ok